### PR TITLE
Popcorn Test

### PR DESCRIPTION
**Pull request by mbohnearschi**
- Generate all possible combinations from letters [O, P, P, R, O, C, N, C]
- Check if the generated dictionary contains the following words:
 [POP, CAT, DOG, ROCK, POPCORN, CORN]

### CODE REVIEW

File: popcorn.rb [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+  def initialize
+    @id = SecureRandom.uuid.to_s
+    @used_letter_ids = []
+    @word = ''
+  end
+end
+
+class Letter
+  attr_accessor :id, :value, :display_name, :neighbours
```
Comment (vlebedeff) [inline popcorn.rb]: Looks like `display_name` is not really used anywhere

File: popcorn.rb [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  end
+
+  def unique_words(word_holder, current_letter)
+    word_list = []
+
+    if word_holder.used_letter_ids.include?(current_letter.id)
+      word_list.push word_holder.word
+    else
+      new_used_letter_ids = word_holder.used_letter_ids.dup
+      new_used_letter_ids.push(current_letter.id)
```
Comment (vlebedeff) [inline popcorn.rb]: `new_used_letter_ids = word_holder.used_letter_ids + [current_letter.id]` will return a new object and is a little easier to read

File: popcorn.rb [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+      new_word = word_holder.word.dup + current_letter.value
```
Comment (vlebedeff) [inline popcorn.rb]: String concatenation `word_holder.word + current_letter.value` already returns a new object. No need to dup

**Comment by mbohnearschi** [comment]
I've updated the implementation to reduce complexity

**Pull request merged**